internal/provider/kubernetes/controller.go

```diff
@@ -909,147 +909,185 @@ func (r *gatewayAPIReconciler) watchResources(ctx context.Context, mgr manager.M
 	// Upon leader election, we retrigger the reconciliation process to allow the elected leader to
 	// process status updates and infrastructure changes. This step is crucial for synchronizing resources
 	// that may have been altered or introduced while there was no elected leader.
-	if err := c.Watch(
-		NewWatchAndReconcileSource(mgr.Elected(), &gwapiv1.GatewayClass{}),
-		handler.EnqueueRequestsFromMapFunc(r.enqueueClass)); err != nil {
+	if err := c.Watch(NewWatchAndReconcileSource(mgr.Elected(), &gwapiv1.GatewayClass{}, handler.EnqueueRequestsFromMapFunc(r.enqueueClass))); err != nil {
 		return err
 	}
 
 	if err := c.Watch(
-		source.Kind(mgr.GetCache(), &gwapiv1.GatewayClass{}),
-		handler.EnqueueRequestsFromMapFunc(r.enqueueClass),
-		predicate.GenerationChangedPredicate{},
-		predicate.NewPredicateFuncs(r.hasMatchingController),
-	); err != nil {
+		source.Kind(mgr.GetCache(), &gwapiv1.GatewayClass{},
+			handler.TypedEnqueueRequestsFromMapFunc(func(ctx context.Context, gc *gwapiv1.GatewayClass) []reconcile.Request {
+				return r.enqueueClass(ctx, gc)
+			}),
+			&predicate.TypedGenerationChangedPredicate[*gwapiv1.GatewayClass]{},
+			predicate.NewTypedPredicateFuncs[*gwapiv1.GatewayClass](r.hasMatchingController))); err != nil {
 		return err
 	}
 
 	// Only enqueue EnvoyProxy objects that match this Envoy Gateway's GatewayClass.
-	epPredicates := []predicate.Predicate{
-		predicate.GenerationChangedPredicate{},
-		predicate.ResourceVersionChangedPredicate{},
```

Comment:
@arkodg is this a must?

Reply:
can you track the orig commit for `ResourceVersionChangedPredicate` ? did the type get removed ?

Reply:
I can find a typed Predicate about that, actually I didn't understand why EG need it only for this.